src/storage/backends/sqlite/SQLite.cc

```diff
@@ -0,0 +1,300 @@
+// See the file "COPYING" in the main distribution directory for copyright.
+
+#include "zeek/storage/backends/sqlite/SQLite.h"
+
+#include "zeek/3rdparty/sqlite3.h"
+#include "zeek/Func.h"
+#include "zeek/Val.h"
+#include "zeek/storage/ReturnCode.h"
+
+namespace zeek::storage::backends::sqlite {
+
+storage::BackendPtr SQLite::Instantiate(std::string_view tag) { return zeek::make_intrusive<SQLite>(tag); }
+
+/**
+ * Called by the manager system to open the backend.
+ *
+ * Derived classes must implement this method. If successful, the
+ * implementation must call \a Opened(); if not, it must call Error()
+ * with a corresponding message.
+ */
+OperationResult SQLite::DoOpen(OpenResultCallback* cb, RecordValPtr options) {
+    if ( sqlite3_threadsafe() == 0 ) {
```

Comment:
Should we even bother to build the plugin if this triggers? We know that all sqlite interactions will always fail if this is the case.

Reply:
I really wish this is something we could check at compile-time via constexpr. I could probably check it via CMake though.

Reply:
I'm going to leave this out of the PR and fix it separately so I can get all of the cases together. I think we can build with thread-safe mode turned on unconditionally, and we don't ever link in an external version ,ti would always be true.

Reply:
Opened https://github.com/zeek/zeek/issues/4285